src/main/scala/com/raphtory/core/components/spout/executor/LiveTwitterSpoutExecutor.scala

```diff
@@ -0,0 +1,130 @@
+package com.raphtory.core.components.spout.executor
+
+import com.raphtory.core.components.spout.SpoutExecutor
+import com.raphtory.core.components.spout.executor.Util.filterRules
+import com.raphtory.core.components.spout.executor.Util.hashtag
+import com.raphtory.core.components.spout.executor.Util.twitterClient
+import com.raphtory.core.components.spout.executor.Util.twitterEventListener
+import com.raphtory.core.config.PulsarController
+import com.raphtory.core.deploy.Raphtory
+import com.raphtory.core.util.FileUtils.logger
+import com.raphtory.serialisers.PulsarKryoSerialiser
+import com.typesafe.config.Config
+import io.github.redouane59.twitter.dto.tweet.Tweet
+import io.github.redouane59.twitter.dto.tweet.TweetType
+import io.github.redouane59.twitter.IAPIEventListener
+import io.github.redouane59.twitter.TwitterClient
+import io.github.redouane59.twitter.dto.rules.FilteredStreamRulePredicate
+import io.github.redouane59.twitter.signature.TwitterCredentials
+import monix.execution.Scheduler
+
+class LiveTwitterSpoutExecutor(
+    conf: Config,
+    pulsarController: PulsarController,
+    scheduler: Scheduler
+) extends SpoutExecutor[Tweet](
+                conf: Config,
+                pulsarController: PulsarController,
+                scheduler: Scheduler
+        ) {
+
+  def streamTweets() =
+    //Filter Twitter stream if hashtag field in application.conf is not empty
+    if (hashtag.nonEmpty) {
+      filterRules()
+      twitterClient.startFilteredStream(twitterEventListener(conf))
+    }
+    else
+      twitterClient.startSampledStream(twitterEventListener(conf))
+
+  override def run(): Unit =
+    streamTweets()
+
+  override def stop(): Unit = ???
```

Comment:
stop(): Unit = twitterClient.stopStream(<.....>)

then you will need to write a stopStream fn